mos-platform/common/c/stdio-full.c

```diff
@@ -128,36 +128,7 @@ static unsigned filemode(const char *const mode) {
   return rc;
 }
 
-static FILE *init_file(FILE *stream) {
-  FILE *rc = stream;
-
-  if (rc == NULL) {
-    if ((rc = (FILE *)malloc(sizeof(struct _FILE))) == NULL) {
-      /* No memory */
-      return NULL;
-    }
-  }
-
-  if ((rc->buffer = (char *)malloc(BUFSIZ)) == NULL) {
-    /* No memory */
-    free(rc);
-    return NULL;
-  }
-
-  rc->bufsize = BUFSIZ;
-  rc->bufidx = 0;
-  rc->bufend = 0;
-  rc->pos = 0;
-  rc->ungetc_buf_full = false;
-  rc->status = FREEBUFFER;
-  rc->del_filename = NULL;
-
-  // TODO: Setting mbstate
-
-  return rc;
-}
-
-static signed char stdio_open(const char *const filename, unsigned int mode) {
+static FILE *stdio_open(FILE *stream, const char *filename, unsigned int mode) {
```

Comment:
This needs a comment describing the relationship between the argument and the return value. In particular, it wasn't clear that this takes ownership of the stream argument, and that that argument must be dynamically allocated.

Reply:
Done.
`stream` is not necessarily `malloc`ed - it can be `stdout` from `freopen`. The old code was doing `free` without checking that - this is one of the bugs I'm fixing.